File: CS/ServerModeWithParameters/AdventureWorksObjects.cs
namespace AdventureWorks {
    using System;
    using DevExpress.Xpo;
    
    
    [Persistent("Sales.SalesOrderHeader")]
    public class OrderHeader : XPLiteObject {
        public OrderHeader(Session session) : base(session) { }

        [Persistent("SalesOrderNumber"), Key]
        string orderNumber = null;

        [PersistentAlias("orderNumber")]
        public string OrderNumber {
            get { return orderNumber; }
        }

        decimal _subTotal;
        public decimal SubTotal {
            get { return _subTotal; }
            set { SetPropertyValue("SubTotal", ref _subTotal, value); }
        }

        DateTime _modifiedDate;
        public DateTime ModifiedDate {
            get { return _modifiedDate; }
            set { SetPropertyValue("ModifiedDate", ref _modifiedDate, value); }
        }

        Territory _territory;
        [Persistent("TerritoryID")]
        public Territory Territory {
            get { return _territory; }
            set { SetPropertyValue("Territory", ref _territory, value); }
        }

        Customer _customer;
        [Persistent("CustomerID")]
        public Customer Customer {
            get { return _customer; }
            set { SetPropertyValue("Customer", ref _customer, value); }
        }

        Address _shipToAddress;
        [Persistent("ShipToAddressID")]
        public Address ShipToAddress {
            get { return _shipToAddress; }
            set { SetPropertyValue("ShipToAddress", ref _shipToAddress, value); }
        }
    }

    [Persistent("Sales.Customer")]
    public class Customer : XPLiteObject {
        public Customer(Session session) : base(session) { }

        [Persistent, Key(true)]
        int CustomerID = 0;

        [PersistentAlias("CustomerID")]
        public int ID {
            get { return CustomerID; }
        }

        Territory _territory;
        [Persistent("TerritoryID"), Association("Territory-Customer")]
        public Territory Territory {
            get { return _territory; }
            set { SetPropertyValue("Territory", ref _territory, value); }
        }

        string _accountNumber;
        public string AccountNumber {
            get { return _accountNumber; }
            set { SetPropertyValue("AccountNumber", ref _accountNumber, value); }
        }

        string _customerType;
        public string CustomerType {
            get { return _customerType; }
            set { SetPropertyValue("CustomerType", ref _customerType, value); }
        }
    }

    [Persistent("Sales.SalesPerson")]
    public class SalesPerson : XPLiteObject {
        public SalesPerson(Session session) : base(session) { }

        [Persistent, Key(true)]
        int SalesPersonID = 0;

        [PersistentAlias("SalesPersonID")]
        public int ID {
            get { return SalesPersonID; }
        }

        Territory _territory;
        [Persistent("TerritoryID")]
        public Territory Territory {
            get { return _territory; }
            set { SetPropertyValue("Territory", ref _territory, value); }
        }
    }

    [Persistent("Sales.SalesTerritory")]
    public class Territory : XPLiteObject {
        public Territory(Session session) : base(session) { }

        [Persistent, Key(true)]
        int TerritoryID = 0;

        [PersistentAlias("TerritoryID")]
        public int ID {
            get { return TerritoryID; }
        }

        string _mame;
        public string Name {
            get { return _mame; }
            set { SetPropertyValue("Name", ref _mame, value); }
        }

        string _group;
        public string Group {
            get { return _group; }
            set { SetPropertyValue("Group", ref _group, value); }
        }

        string _countryRegionCode;
        public string CountryRegionCode {
            get { return _countryRegionCode; }
            set { SetPropertyValue("CountryRegionCode", ref _countryRegionCode, value); }
        }

        [Association("Territory-Customer")]
        public XPCollection<Customer> Customers {
            get { return GetCollection<Customer>("Customers"); }
        }

        [Association("Territory-States")]
        public XPCollection<StateProvince> States {
            get { return GetCollection<StateProvince>("States"); }
        }
    }

    [Persistent("Person.StateProvince")]
    public class StateProvince : XPLiteObject {
        public StateProvince(Session session) : base(session) { }

        [Persistent, Key(true)]
        int StateProvinceID = 0;

        [PersistentAlias("StateProvinceID")]
        public int ID {
            get { return StateProvinceID; }
        }

        string _name;
        public string Name {
            get { return _name; }
            set { SetPropertyValue("Name", ref _name, value); }
        }

        string _stateCode;
        [Persistent("StateProvinceCode")]
        public string StateCode {
            get { return _stateCode; }
            set { SetPropertyValue("StateCode", ref _stateCode, value); }
        }

        Territory _territory;
        [Persistent("TerritoryID"), Association("Territory-States")]
        public Territory Territory {
            get { return _territory; }
            set { SetPropertyValue("Territory", ref _territory, value); }
        }

        [Association("State-Addresses")]
        public XPCollection<Address> Addresses {
            get { return GetCollection<Address>("Addresses"); }
        }
    }

    [Persistent("Person.Address")]
    public class Address : XPLiteObject {
        public Address(Session session) : base(session) { }

        [Persistent, Key(true)]
        int AddressID = 0;

        [PersistentAlias("AddressID")]
        public int ID {
            get { return AddressID; }
        }

        StateProvince _state;
        [Persistent("StateProvinceID"), Association("State-Addresses")]
        public StateProvince State {
            get { return _state; }
            set { SetPropertyValue("State", ref _state, value); }
        }

        string _city;
        public string City {
            get { return _city; }
            set { SetPropertyValue("City", ref _city, value); }
        }


        string _addressLine1;
        public string AddressLine1 {
            get { return _addressLine1; }
            set { SetPropertyValue("AddressLine1", ref _addressLine1, value); }
        }
    }
}

File: VB/ServerModeWithParameters/AdventureWorksObjects.vb
Imports Microsoft.VisualBasic
	Imports System
	Imports DevExpress.Xpo
Namespace AdventureWorks


	<Persistent("Sales.SalesOrderHeader")> _
	Public Class OrderHeader
		Inherits XPLiteObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		<Persistent("SalesOrderNumber"), Key> _
		Private orderNumber_Renamed As String = Nothing

		<PersistentAlias("orderNumber")> _
		Public ReadOnly Property OrderNumber() As String
			Get
				Return orderNumber_Renamed
			End Get
		End Property

		Private _subTotal As Decimal
		Public Property SubTotal() As Decimal
			Get
				Return _subTotal
			End Get
			Set(ByVal value As Decimal)
				SetPropertyValue("SubTotal", _subTotal, value)
			End Set
		End Property

		Private _modifiedDate As DateTime
		Public Property ModifiedDate() As DateTime
			Get
				Return _modifiedDate
			End Get
			Set(ByVal value As DateTime)
				SetPropertyValue("ModifiedDate", _modifiedDate, value)
			End Set
		End Property

		Private _territory As Territory
		<Persistent("TerritoryID")> _
		Public Property Territory() As Territory
			Get
				Return _territory
			End Get
			Set(ByVal value As Territory)
				SetPropertyValue("Territory", _territory, value)
			End Set
		End Property

		Private _customer As Customer
		<Persistent("CustomerID")> _
		Public Property Customer() As Customer
			Get
				Return _customer
			End Get
			Set(ByVal value As Customer)
				SetPropertyValue("Customer", _customer, value)
			End Set
		End Property

		Private _shipToAddress As Address
		<Persistent("ShipToAddressID")> _
		Public Property ShipToAddress() As Address
			Get
				Return _shipToAddress
			End Get
			Set(ByVal value As Address)
				SetPropertyValue("ShipToAddress", _shipToAddress, value)
			End Set
		End Property
	End Class

	<Persistent("Sales.Customer")> _
	Public Class Customer
		Inherits XPLiteObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		<Persistent, Key(True)> _
		Private CustomerID As Integer = 0

		<PersistentAlias("CustomerID")> _
		Public ReadOnly Property ID() As Integer
			Get
				Return CustomerID
			End Get
		End Property

		Private _territory As Territory
		<Persistent("TerritoryID"), Association("Territory-Customer")> _
		Public Property Territory() As Territory
			Get
				Return _territory
			End Get
			Set(ByVal value As Territory)
				SetPropertyValue("Territory", _territory, value)
			End Set
		End Property

		Private _accountNumber As String
		Public Property AccountNumber() As String
			Get
				Return _accountNumber
			End Get
			Set(ByVal value As String)
				SetPropertyValue("AccountNumber", _accountNumber, value)
			End Set
		End Property

		Private _customerType As String
		Public Property CustomerType() As String
			Get
				Return _customerType
			End Get
			Set(ByVal value As String)
				SetPropertyValue("CustomerType", _customerType, value)
			End Set
		End Property
	End Class

	<Persistent("Sales.SalesPerson")> _
	Public Class SalesPerson
		Inherits XPLiteObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		<Persistent, Key(True)> _
		Private SalesPersonID As Integer = 0

		<PersistentAlias("SalesPersonID")> _
		Public ReadOnly Property ID() As Integer
			Get
				Return SalesPersonID
			End Get
		End Property

		Private _territory As Territory
		<Persistent("TerritoryID")> _
		Public Property Territory() As Territory
			Get
				Return _territory
			End Get
			Set(ByVal value As Territory)
				SetPropertyValue("Territory", _territory, value)
			End Set
		End Property
	End Class

	<Persistent("Sales.SalesTerritory")> _
	Public Class Territory
		Inherits XPLiteObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		<Persistent, Key(True)> _
		Private TerritoryID As Integer = 0

		<PersistentAlias("TerritoryID")> _
		Public ReadOnly Property ID() As Integer
			Get
				Return TerritoryID
			End Get
		End Property

		Private _mame As String
		Public Property Name() As String
			Get
				Return _mame
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _mame, value)
			End Set
		End Property

		Private _group As String
		Public Property Group() As String
			Get
				Return _group
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Group", _group, value)
			End Set
		End Property

		Private _countryRegionCode As String
		Public Property CountryRegionCode() As String
			Get
				Return _countryRegionCode
			End Get
			Set(ByVal value As String)
				SetPropertyValue("CountryRegionCode", _countryRegionCode, value)
			End Set
		End Property

		<Association("Territory-Customer")> _
		Public ReadOnly Property Customers() As XPCollection(Of Customer)
			Get
				Return GetCollection(Of Customer)("Customers")
			End Get
		End Property

		<Association("Territory-States")> _
		Public ReadOnly Property States() As XPCollection(Of StateProvince)
			Get
				Return GetCollection(Of StateProvince)("States")
			End Get
		End Property
	End Class

	<Persistent("Person.StateProvince")> _
	Public Class StateProvince
		Inherits XPLiteObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		<Persistent, Key(True)> _
		Private StateProvinceID As Integer = 0

		<PersistentAlias("StateProvinceID")> _
		Public ReadOnly Property ID() As Integer
			Get
				Return StateProvinceID
			End Get
		End Property

		Private _name As String
		Public Property Name() As String
			Get
				Return _name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _name, value)
			End Set
		End Property

		Private _stateCode As String
		<Persistent("StateProvinceCode")> _
		Public Property StateCode() As String
			Get
				Return _stateCode
			End Get
			Set(ByVal value As String)
				SetPropertyValue("StateCode", _stateCode, value)
			End Set
		End Property

		Private _territory As Territory
		<Persistent("TerritoryID"), Association("Territory-States")> _
		Public Property Territory() As Territory
			Get
				Return _territory
			End Get
			Set(ByVal value As Territory)
				SetPropertyValue("Territory", _territory, value)
			End Set
		End Property

		<Association("State-Addresses")> _
		Public ReadOnly Property Addresses() As XPCollection(Of Address)
			Get
				Return GetCollection(Of Address)("Addresses")
			End Get
		End Property
	End Class

	<Persistent("Person.Address")> _
	Public Class Address
		Inherits XPLiteObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		<Persistent, Key(True)> _
		Private AddressID As Integer = 0

		<PersistentAlias("AddressID")> _
		Public ReadOnly Property ID() As Integer
			Get
				Return AddressID
			End Get
		End Property

		Private _state As StateProvince
		<Persistent("StateProvinceID"), Association("State-Addresses")> _
		Public Property State() As StateProvince
			Get
				Return _state
			End Get
			Set(ByVal value As StateProvince)
				SetPropertyValue("State", _state, value)
			End Set
		End Property

		Private _city As String
		Public Property City() As String
			Get
				Return _city
			End Get
			Set(ByVal value As String)
				SetPropertyValue("City", _city, value)
			End Set
		End Property


		Private _addressLine1 As String
		Public Property AddressLine1() As String
			Get
				Return _addressLine1
			End Get
			Set(ByVal value As String)
				SetPropertyValue("AddressLine1", _addressLine1, value)
			End Set
		End Property
	End Class
End Namespace
